protocol/op-stack-manager-arch.md

```diff
@@ -0,0 +1,294 @@
+
+Each step is a separate forge script that takes the input TOML file as the only input, and outputs a TOML file.
+To make the artifact self-contained, the outputs also include the inputs.
+This allows a single artifact to be used to convey all information needed about a deployment.
+
+These forge scripts can be wrapped with go code to facilitate e2e testing by the interop team.
+The go code should leverage the same file-based interface to ensure equivalence between the forge script execution and the go execution.
+
+## Step 1: Deploy Superchain contracts
+
+The sub-steps are:
+
+1. Deploy a `ProxyAdmin`.
+2. Deploy a `Proxy` owned by `ProxyAdmin`, set the `SuperchainConfig` contract as its implementation, and initialize it.
+3. Deploy a `Proxy` owned by `ProxyAdmin`, set the `ProtocolVersions` contract as its implementation, and initialize it.
+
+The inputs for each sub-step are:
+
+1. A `proxyAdminOwner`.
+2. The Guardian address and the initial pause status. (The admin of its `Proxy` is always the `ProxyAdmin` from sub-step 1, so this is not an input).
+3. A `protocolVersionsOwner`, the `requiredVersion`, and the `recommendedVersion`. (The admin of its `Proxy` is always the `ProxyAdmin` from sub-step 1, so this is not an input).
+
+The outputs for each sub-step are:
+
+1. The `superchainProxyAdmin` contract address.
+2. The `superchainConfigProxy` and `superchainConfigImplementation` contract addresses.
+3. The `protocolVersionsProxy` and `protocolVersionsImplementation` contract addresses.
+
+A sample TOML deploy config input for this step might look like:
+
+```toml
+[roles]
+proxyAdminOwner = "0x1234..."
+protocolVersionsOwner = "0x1234..."
+
+[superchain_config]
+guardian = "0x1234..."
+paused = false
+
+[protocol_versions]
+requiredVersion = "0.1.0"
+recommendedVersion = "0.1.0"
+```
+
+A sample TOML output for this step might look like:
+
+```toml
+superchainProxyAdmin = "0x1234..."
+superchainConfigProxy = "0x1234..."
+superchainConfigImpl = "0x1234..."
+protocolVersionsProxy = "0x1234..."
+protocolVersionsImpl = "0x1234..."
+
+# Everything above this comment are the outputs, below are the inputs.
+[roles]
+proxyAdminOwner = "0x1234..."
+protocolVersionsOwner = "0x1234..."
+
+[superchain_config]
+guardian = "0x1234..."
+paused = false
+
+[protocol_versions]
+requiredVersion = "0.1.0"
+recommendedVersion = "0.1.0"
+```
+
+## Step 2: Deploy Shared Implementation Contracts
+
+We assume the latest Fault Proofs release `op-contracts/v1.5.0`.
+The exact set of sub-steps and inputs may change with future releases.
+
+The sub-steps are deploying the following contracts:
+
+1. DisputeGameFactory
+2. AnchorStateRegistry
+3. DelayedWETH
+4. PreimageOracle
+5. MIPS
+6. OptimismPortal
+7. SystemConfig
+8. L1CrossDomainMessenger
+9. L1ERC721Bridge
+10. L1StandardBridge
+11. OptimismMintableERC20Factory
+
+The inputs for each sub-step are:
+
+1. None
```

Comment:
Will `DGF.setImplementation()` need to be called? If so that will require a dispute game impl, which is not yet MCP ready AFAIR.